app/frontend/package.json

```diff
@@ -16,6 +16,7 @@
     "maplibre-gl": "^1.14.0",
     "react": "^17.0.0",
     "react-dom": "^17.0.0",
+    "react-error-boundary": "^3.1.1",
```

Comment:
I'm debating with myself if it's worth adding a dependency for this.

The pro is our code can be cleaner, but the con is we are then relying on this package being maintained. Considering we are hoping to support the app long-term, we shouldn't add dependencies too lightly, and it's not that difficult to manually do an [error boundary component](https://reactjs.org/docs/error-boundaries.html).

What do you think?

Reply:
It's a small dependency with many weekly downloads and no other dependencies except for @babel/runtime which I raised a question for on the react-error-boundary repo to know why they have that dependency.
Making your own error boundary is a possibility and is fairly simple indeed but doesn't mean it won't need maintenance in the future. I would say that, if necessary, we can make the ErrorBoundary ourselves once we notice problems with maintenance of react-error-boundary.